### PR TITLE
New EasyConfigs for LLVM with the gimkl-2017a toolchain

### PR DESCRIPTION
**Pull request by valtandor**
LLVM 4.0.0 requires a Python build (WIP).

### CODE REVIEW

**Comment by boegel** [comment]
@valtandor Is there another PR for the missing Python dependency?

**Comment by valtandor** [comment]
@boegel I would welcome your advice on how to resolve a circular dependency, i.e., LLVM requires Python, and the Python easyconfig I have prepared requires LLVM for some of its dependencies. Is the best thing to submit a stripped-down Python easyconfig without the LLVM-requiring parts?

File: easybuild/easyconfigs/l/LLVM/LLVM-3.8.1-gimkl-2017a.eb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ specified code representation known as the LLVM intermediate representation
+ ("LLVM IR"). The LLVM Core libraries are well documented, and it is
+ particularly easy to invent your own language (or port an existing compiler)
+ to use LLVM as an optimizer and code generator."""
+
+toolchain = {'name': 'gimkl', 'version': '2017a'}
+toolchainopts = {'pic': True, 'cstd': 'c++14'}
+
+source_urls = ["http://llvm.org/releases/%(version)s"]
+sources = ["llvm-%(version)s.src.tar.xz"]
```
Comment (verdurin) [inline easybuild/easyconfigs/l/LLVM/LLVM-3.8.1-gimkl-2017a.eb]: @valtandor Please add a SHA256 checksum

File: easybuild/easyconfigs/l/LLVM/LLVM-4.0.0-gimkl-2017a.eb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ ("LLVM IR"). The LLVM Core libraries are well documented, and it is
+ particularly easy to invent your own language (or port an existing compiler)
+ to use LLVM as an optimizer and code generator."""
+
+toolchain = {'name': 'gimkl', 'version': '2017a'}
+toolchainopts = {'pic': True, 'cstd': 'c++14'}
+
+source_urls = ["http://llvm.org/releases/%(version)s"]
+sources = ["llvm-%(version)s.src.tar.xz"]
+patches = ["LLVM-4.0.0-fix-shlib.patch"]
```
Comment (verdurin) [inline easybuild/easyconfigs/l/LLVM/LLVM-4.0.0-gimkl-2017a.eb]: @valtandor Please add a checksum here too.

**Comment by migueldiascosta** [comment]
@valtandor, this PR is being closed for the following reason(s): no activity for > 1 year.
Please don't hesitate to reopen this PR or add a comment if you feel this contribution is still relevant.
For more information on our policy w.r.t. closing PRs, see https://easybuild.readthedocs.io/en/latest/Contributing.html#why-a-pull-request-may-be-closed-by-a-maintainer

**Pull request closed without merging**